core/src/main/java/com/valtech/aem/saas/core/indexing/DefaultIndexUpdateService.java

```diff
@@ -0,0 +1,154 @@
+package com.valtech.aem.saas.core.indexing;
+
+import com.google.gson.Gson;
+import com.valtech.aem.saas.api.indexing.IndexContentPayload;
+import com.valtech.aem.saas.api.indexing.IndexUpdateResponse;
+import com.valtech.aem.saas.api.indexing.IndexUpdateService;
+import com.valtech.aem.saas.core.http.client.SearchRequestExecutorService;
+import com.valtech.aem.saas.core.http.client.SearchServiceConnectionConfigurationService;
+import com.valtech.aem.saas.core.http.request.SearchRequest;
+import com.valtech.aem.saas.core.http.request.SearchRequestDelete;
+import com.valtech.aem.saas.core.http.request.SearchRequestPost;
+import com.valtech.aem.saas.core.http.response.DefaultIndexUpdateDataExtractionStrategy;
+import com.valtech.aem.saas.core.indexing.DefaultIndexUpdateService.Configuration;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.entity.EntityBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.message.BasicNameValuePair;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@Slf4j
+@Component(name = "Search as a Service - Index Update Service",
+    service = IndexUpdateService.class)
+@Designate(ocd = Configuration.class)
+public class DefaultIndexUpdateService implements IndexUpdateService {
+
+  public static final String REQUEST_PARAMETER_URL = "url";
+  public static final String REQUEST_PARAMETER_REPOSITORY_PATH = "repository_path";
+
+  @Reference
+  private SearchRequestExecutorService searchRequestExecutorService;
+
+  @Reference
+  private SearchServiceConnectionConfigurationService searchServiceConnectionConfigurationService;
+
+  private Configuration configuration;
+
+  @Activate
+  @Modified
+  private void activate(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  @Override
+  public Optional<IndexUpdateResponse> indexUrl(@NonNull String client, @NonNull String url,
+      @NonNull String repositoryPath) {
+    validateInput(client, url, repositoryPath);
+
+    SearchRequest searchRequest = SearchRequestPost.builder()
+        .uri(getRequestUri(client, configuration.indexUpdateService_apiIndexTriggerAction()))
+        .httpEntity(EntityBuilder.create().setParameters(new BasicNameValuePair(REQUEST_PARAMETER_URL, url),
```

Comment:
it is very difficult to read this builder, personally, i would extract some of these calls in variables first and than use them here